packages/collection/src/collection.ts

```diff
@@ -148,10 +162,21 @@ export class Collection<Key, Value> extends Map<Key, Value> {
 	 *
 	 * @param index - The index of the element to obtain
 	 */
-	public at(index: number) {
-		index = Math.floor(index);
-		const arr = [...this.values()];
-		return arr.at(index);
+	public at(index: number): Value | undefined {
+		index = Math.trunc(index);
+		if (index >= 0) {
+			if (index >= this.size) return undefined;
+		} else {
+			if (index < this.size * -1) return undefined;
+			index += this.size;
+		}
```

Comment:
Less checks, better readability
```suggestion
		if (index >= this.size) return undefined;
		if (index < 0) {
			index += this.size;
			if (index < 0) return undefined;
		}
```

Reply:
I think making one addition is faster then multiplication + addition

Reply:
Yeah, it's 14 8% faster for negative indexes

Reply:
Addition is always going to be faster than multiplication, indeed

Reply:
It's a single operation per method call rather than something involved in the iteration, so the cost is negligible either way. Originally written like this so that the behaviour is obvious at a glance, but there is scope to tweak it.